pkg/virtual/framework/dynamic/apidefs/types.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apidefs
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/handlers"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	apiresourcev1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apiresource/v1alpha1"
+)
+
+// APIDomainKeyContextKeyType is the type of the key for the request context value
+// that will carry the API domain key.
+type APIDomainKeyContextKeyType string
+
+// APIDomainKeyContextKey is the key for the request context value
+// that will carry the API domain key.
+const APIDomainKeyContextKey APIDomainKeyContextKeyType = "VirtualWorkspaceAPIDomainLocationKey"
+
+// APIDefinition provides access to all the information needed to serve a given API resource
+type APIDefinition interface {
+	// GetAPIResourceSpec provides the API resource specification, which contains the
+	// API names, sub-resource definitions, and the OpenAPIv3 schema.
+	GetAPIResourceSpec() *apiresourcev1alpha1.CommonAPIResourceSpec
```

Comment:
can this only be backed by APIResources?

Reply:
looked further down at the forked handler code it looks like returning a CRD here would make the handler code much more similar to apiextensions-apiserver, potentially even allow to reuse the handler.

Reply:
cc @ncdc

Reply:
The main point is that we don't need all the multi-version aspects here.

Reply:
But I assume we could also wrap this info into a CRD if we want to reuse code. We've been discussing the 2 options a while ago.

Reply:
We discussed this and will postpone until the APIBinding refactor.